src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -358,13 +371,6 @@
         ),
     ),
     FrameClauseUnitGrammar=OneOf("RANGE", "ROWS", "GROUPS"),
-    # In Postgres, column references may be followed by a time zone cast in all cases.
-    # For more information, see
-    # https://www.postgresql.org/docs/11/functions-datetime.html
-    ColumnReferenceSegment=Sequence(
-        ansi.ColumnReferenceSegment,
-        Ref("ArrayAccessorSegment", optional=True),
-    ),
```

Comment:
Moved this into the class. This does change the parse tree slightly but couldn't get it to work like it did previously (and it was confusing as it was).

Reply:
Actually removed this completely as was only needed for the `SET` command so added it there instead.